### PR TITLE
Update lesson

### PR DESCRIPTION
**Pull request by jlboba**
Changes made:

- Config for TS
- Added small section on typing state and updated examples accordingly. Otherwise, left this lesson mostly untouched

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,6 +106,20 @@ We can then use the `count` variable to display its current value in the
...
+## Typing State 
+
+When initially setting state, TypeScript can infer the type to use based on
+the initial value. However, if it's desired to be more explicit about the type, 
+we can do so as well using the generic syntax `<>`. For example with our `count`
+state, we could write: 
+
+`const [count, setCount] = useState<number>(0);` 
+
+It may be tempting to type within the descrutctured syntax, but just as we are
```
Comment (lizbur10) [inline README.md]: Typo: descrutctured

File: README.md [hunk 2]
```diff
@@ -106,6 +106,20 @@ We can then use the `count` variable to display its current value in the
 // => <button>0</button>
 ```
 
+## Typing State 
+
+When initially setting state, TypeScript can infer the type to use based on
+the initial value. However, if it's desired to be more explicit about the type,
```
Comment (lizbur10) [inline README.md]: Is one approach preferred over the other?

File: README.md [hunk 3]
```diff
@@ -106,6 +106,20 @@ We can then use the `count` variable to display its current value in the
...
+
+When initially setting state, TypeScript can infer the type to use based on
+the initial value. However, if it's desired to be more explicit about the type, 
+we can do so as well using the generic syntax `<>`. For example with our `count`
+state, we could write: 
+
+`const [count, setCount] = useState<number>(0);` 
+
+It may be tempting to type within the descrutctured syntax, but just as we are 
+unable to type within destructured objects, we cannot do so within destructured
```
Comment (lizbur10) [inline README.md]: Recommended rewording:

"You may recall from an earlier lesson that typing is not allowed within destructured objects; the same applies to destructured arrays as well. Typing must instead be done using the generic syntax, which will handle ..."

File: README.md [hunk 4]
```diff
@@ -168,7 +182,7 @@ component (pay close attention to the `console.log()`s:
```
Comment (lizbur10) [inline README.md:193]: Missing end parenthesis on the line above

File: README.md [hunk 5]
```diff
@@ -300,7 +314,7 @@ docs][rules of hooks explanation].
 
 #### Only Call Hooks from React Functions
 
-> Don’t call Hooks from regular JavaScript functions.
+> Don’t call Hooks from regular TypeScript functions.
```
Comment (lizbur10) [inline README.md]: I would recommend using "JavaScript or TypeScript" here

**Review by lizbur10 — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -115,7 +140,7 @@ case, `setCount`):
 
 ```jsx
 function Counter() {
-  const [count, setCount] = useState(0);
+  const [count, setCount] = useState < number > 0;
```
Comment (jlboba) [inline README.md]:
```suggestion
  const [count, setCount] = useState<number>(0);
```

File: README.md [hunk 7]
```diff
@@ -168,7 +193,7 @@ component (pay close attention to the `console.log()`s:
...
```
Comment (jlboba) [inline README.md]:
```suggestion
  const [count, setCount] = useState<number>(0);
```

File: README.md [hunk 8]
```diff
@@ -203,7 +228,7 @@ counter). To demonstrate the issue, consider the following:
...
```
Comment (jlboba) [inline README.md]:
```suggestion
  const [count, setCount] = useState<number>(0);
```

File: README.md [hunk 9]
```diff
@@ -282,10 +307,10 @@ is called every time your component is rendered. That means this syntax isn't
 valid:
 
 ```jsx
-function Counter(props) {
+function Counter(props: { shouldHaveCount: boolean }) {
   if (props.shouldHaveCount) {
     // This is wrong -- never call a hook inside a condition
-    const [count, setCount] = useState(0);
+    const [count, setCount] = useState < number > 0;
```
Comment (jlboba) [inline README.md]:
```suggestion
    const [count, setCount] = useState<number>(0);
```

**Pull request merged**